js/dcf-modal.js

```diff
@@ -129,7 +129,16 @@ export class DCFModal {
 
     // Trigger pre-open modal event for this modal to allow event listeners to handle
     const preOpenEventName = `ModalPreOpenEvent_${modalId}`;
-    document.dispatchEvent(new CustomEvent(preOpenEventName));
+    const openButtonElement = document.getElementById(openBtnId);
+    let myevent = new CustomEvent(preOpenEventName);
+    if(openButtonElement){
```

Comment:
Can you add some comments to explain what openButtonElement is and why you are doing the if statement. Just for maintainability for future developers.